src/compiler/parser.ts

```diff
@@ -4968,14 +5113,15 @@ module ts {
 
         function parseDeclaration(): ModuleElement {
             let fullStart = getNodePos();
+            let decorators = parseDecorators();

```

Comment:
Subscribers, discussion is being had here https://github.com/wycats/javascript-decorators/issues/7#issuecomment-86247666